### PR TITLE
Removes a bunch of spawn()'s

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
Refactors code to use callbacks and/or async invokes, etc. over using `spawn()`s. Tested everything locally as best I could to be sure, didn't see any issues, but this may need a TM anyway.

## Why It's Good For The Game
Code improvement because spawn()'s are bad practice.

## Changelog
N/A

### CODE REVIEW

File: code/_globalvars/misc.dm [hunk 1]
```diff
@@ -5,20 +5,8 @@ GLOBAL_VAR_INIT(CELLRATE, 0.002)  // conversion ratio between a watt-tick and ki
...
-  spawn(0)
-    GLOB.global_announcer = new(null)
-  return
-
-/proc/create_command_announcer()
-  spawn(0)
-    GLOB.command_announcer = new(null)
-  return
+GLOBAL_DATUM_INIT(global_announcer, /obj/item/radio/intercom, new(null))
+GLOBAL_DATUM_INIT(command_announcer, /obj/item/radio/intercom/command, new(null))
```
Comment (AffectedArc07) [inline code/_globalvars/misc.dm:9]: I tried to convert these, got runtimes on startup. I wish you the best of luck.

File: code/controllers/subsystem/ticker.dm [hunk 2]
```diff
@@ -123,14 +123,14 @@ SUBSYSTEM_DEF(ticker)
...
-			spawn(50)
-				if(mode.station_was_nuked)
-					world.Reboot("Station destroyed by Nuclear Device.", "nuke")
-				else
-					world.Reboot("Round ended.", "proper completion")
+/datum/controller/subsystem/ticker/proc/call_reboot()
+	if(mode.station_was_nuked)
+		world.Reboot("Station destroyed by Nuclear Device.", "end_proper", "nuke")
+	else
+		world.Reboot("Round ended.", "end_proper", "proper completion")
```
Comment (AffectedArc07) [inline code/controllers/subsystem/ticker.dm:133]: This has been conflicted with the SSblackbox PR. Please update and normalise back.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/ticker.dm (hunk 2 above)
Comment (SteelSlayer) [reply code/controllers/subsystem/ticker.dm:133]: What do you mean it has been conflicted?

File: code/_globalvars/misc.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/_globalvars/misc.dm:9]: Hm, I double checked again just now, and I didn't get any runtimes, but I'm on local so yknow.

File: code/controllers/subsystem/ticker.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/controllers/subsystem/ticker.dm:133]: The args are now
```dm
/world/Reboot(var/reason, end_string, var/time)
```

When making this, you re-added the `"end proper"` string, which isnt valid anymore.

Comment (SteelSlayer) [reply code/controllers/subsystem/ticker.dm:133]: Ah gotcha.

File: code/_globalvars/misc.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/_globalvars/misc.dm:9]: Ok so this actually runtimes in the DD window, not in the runtime viewer or logs. Yet AI arrival announcements worked fine when I tested them in game. But I'm probably going to revert this change, don't want to deal with this mess right now.

File: code/datums/diseases/advance/symptoms/shedding.dm [hunk 3]
```diff
@@ -36,15 +36,16 @@ BONUS
...
 						to_chat(H, "<span class='warning'>Your hair starts to fall out in clumps...</span>")
-						spawn(50)
-							head_organ.f_style = "Shaved"
-							head_organ.h_style = "Bald"
-							H.update_hair()
-							H.update_fhair()
-	return
+						addtimer(CALLBACK(src, .proc/change_hair, H, head_organ, "Shaved", "Bald"), 5 SECONDS)
+
+/datum/symptom/shedding/proc/change_hair(mob/living/carbon/human/H, obj/item/organ/external/head/head_organ, f_style, h_style)
```
Comment (Fox-McCloud) [inline code/datums/diseases/advance/symptoms/shedding.dm:45]: make sure to check for the head organ, again---and `H`; they could have gotten deleted.

File: code/datums/helper_datums/teleport.dm [hunk 4]
```diff
@@ -84,14 +84,12 @@
 /datum/teleport/proc/playSpecials(atom/location,datum/effect_system/effect,sound)
 	if(location)
 		if(effect)
-			spawn(-1)
```
Comment (Fox-McCloud) [inline code/datums/helper_datums/teleport.dm:87]: what even *why*

File: code/game/machinery/alarm.dm [hunk 5]
```diff
@@ -1047,8 +1047,7 @@
 		stat &= ~NOPOWER
 	else
 		stat |= NOPOWER
-	spawn(rand(0,15))
```
Comment (Fox-McCloud) [inline code/game/machinery/alarm.dm:1050]: I get what they were going for here---having a staggered effect of them turning off at random intervals, but....like, you're rarely going to notice this.

It's really fascinating to see the attention to detail some old coders had for small things like this, yet the detail to code quality was atrocious.

File: code/game/objects/items/devices/whistle.dm [hunk 6]
```diff
@@ -6,11 +9,11 @@
 	item_state = "flashtool"	//looks exactly like a flash (and nothing like a flashbang)
 	w_class = WEIGHT_CLASS_TINY
 	flags = CONDUCT
-
+	var/next_use_time
 	var/spamcheck = 0
 
 /obj/item/hailer/attack_self(mob/living/carbon/user as mob)
-	if(spamcheck)
+	if(world.time < next_use_time)
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/whistle.dm:16]: my favorite time of timers.

**Comment by Fox-McCloud** [comment]
Just one thing about the head, and I think this'll be good to go.

Sorry about the atmos conflicts.

File: code/game/machinery/alarm.dm (hunk 5 above)
Comment (SteelSlayer) [reply code/game/machinery/alarm.dm:1050]: Yeah, that's how I felt about this. It's like no one's going to care or notice this.

File: code/datums/diseases/advance/symptoms/shedding.dm (hunk 3 above)
Comment (SteelSlayer) [reply code/datums/diseases/advance/symptoms/shedding.dm:45]: Can do.

**Comment by SteelSlayer** [comment]
Conflicts resolved, changes made. Should be good now.

**Review by dearmochi — APPROVED** [review]

**Pull request merged**